### ISSUE TITLE
False positive case of no-noninteractive-element-interactions rule for popup modals.

### ISSUE THREAD

**Issue by tranvansang** [jsx-eslint/eslint-plugin-jsx-a11y#781]
I am implementing a popup modal that listens to the escape key to close itself.

In other cases, I also implement a collapsing component which I want it to collapse when the user presses the Esc key.

```
<button onClick={() => setShow(s => !s)}>Toggle</button>
<div onKeyDown={evt => {
  if (['Esc', 'Escape'].includes(evt.key)) {
    evt.stopPropagation()
    setShow(false)
  }
}>
  modal content
</div>
```

The `no-noninteractive-element-interactions` rule always complains about this snippet and it is quite noisy in my project.

[This](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-element-interactions.md#case-this-element-is-catching-bubbled-events-from-elements-that-it-contains) suggests the usage of the `role="presentation"`. However, does this also disable the semantic meaning of the whole element's children?
